src/core/digitizinglogger.cpp

```diff
@@ -0,0 +1,235 @@
+  ---------------------
+ begin                : 7.6.2021
+ copyright            : (C) 2021 by Mathieu Pellerin
+ email                : mathieu (at) opengis.ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "digitizinglogger.h"
+#include "expressioncontextutils.h"
+
+#include <qgsexpressioncontextutils.h>
+#include <qgslayertree.h>
+#include <qgsmessagelog.h>
+#include <qgsvectorlayerutils.h>
+#include <qgswkbtypes.h>
+
+#include <QDateTime>
+
+DigitizingLogger::DigitizingLogger()
+{
+}
+
+void DigitizingLogger::setType( const QString &type )
+{
+  if ( mType == type )
+    return;
+
+  mType = type;
+
+  emit typeChanged();
+}
+
+void DigitizingLogger::setPositionInformation( const GnssPositionInformation &positionInformation )
+{
+  mPositionInformation = positionInformation;
+
+  emit positionInformationChanged();
+}
+
+void DigitizingLogger::setPositionLocked( bool positionLocked )
+{
+  if ( mPositionLocked == positionLocked )
+    return;
+
+  mPositionLocked = positionLocked;
+
+  emit positionLockedChanged();
+}
+
+void DigitizingLogger::setTopSnappingResult( const SnappingResult &topSnappingResult )
+{
+  mTopSnappingResult = topSnappingResult;
+
+  emit topSnappingResultChanged();
+}
+
+void DigitizingLogger::setProject( QgsProject *project )
+{
+  if ( mProject == project )
+    return;
+
+  if ( mProject )
+  {
+    disconnect( mProject, &QgsProject::readProject, this, &DigitizingLogger::findLogsLayer );
+  }
+
+  mProject = project;
+  connect( mProject, &QgsProject::readProject, this, &DigitizingLogger::findLogsLayer );
+
+  clearCoordinates();
+  findLogsLayer();
+
+  emit projectChanged();
+}
+
+void DigitizingLogger::setMapSettings( QgsQuickMapSettings *mapSettings )
+{
+  if ( mMapSettings == mapSettings )
+    return;
+
+  mMapSettings = mapSettings;
+
+  emit mapSettingsChanged();
+}
+
+void DigitizingLogger::setDigitizingLayer( QgsVectorLayer *layer )
+{
+  if ( mDigitizingLayer == layer )
+    return;
+
+  mDigitizingLayer = layer;
+
+  emit digitizingLayerChanged();
+}
+
+void DigitizingLogger::findLogsLayer()
+{
+  mLogsLayer = nullptr;
+  if ( mProject )
+  {
+    const QString logsLayerId = mProject->readEntry( QStringLiteral( "qfieldsync" ), QStringLiteral( "digitizingLogsLayer" ) );
+    if ( !logsLayerId.isEmpty() )
+    {
+      QgsLayerTreeLayer *item = mProject->layerTreeRoot()->findLayer( logsLayerId );
```

Comment:
More of a qf sync question: do we always have the offline id here and never the online id?